app/modelmodule/admin_acls.py

```diff
@@ -20,11 +20,19 @@ class AdminPermission(db.Model):
     class AccessLevel(object):
         """An enum for the level of access the user has."""
 
-        # Admin access: the admin can edit the repo, add new moderators, etc.
-        ADMINISTRATOR = 0
-        # Moderator access: the admin can moderate user input, but not anything
-        # else.
-        MODERATOR = 1
+        # Superadmins can do anything for a repo, and global superadmins can
+        # create new repos.
+        SUPERADMIN = 0
+        # Managers have limited administrative access: e.g., they can edit
+        # custom messages for their repo, but not the title.
+        MANAGER = 1
+        # Moderators can moderate user input.
+        MODERATOR = 2
+
+        # An ordered list of the admin access levels, from lowest to highest
+        # (i.e., each level has the permissions of everything before it in the
+        # list).
+        ORDERING = [MODERATOR, MANAGER, SUPERADMIN]
```

Comment:
I'm clearly overthinking this - just curious if it's better to rely just on the integer value of the enum (which is used in the template {% if acl.access_level == 0 %}) instead of the array? It looks like duplicated information, and I'm not sure what would happen if either the enum values or the ordering changed in the future?

Reply:
I want the order to be independent of the enum values because the enum values are going in the database. E.g., if we later want to add a new level between manager and superadmin, we can make its value 3 but just put it in the right place in `ORDERING` and have everything work.